TV Show Project: Level 100-400

kmh256 opened the pull request:
Project complete up to Level 400

kmh256 committed:
Personalised Project Title

kmh256 committed:
Lesson 100 - Added cards

Added bootstrap cards, episode headings, images and summaries.

kmh256 committed:
Level 100 Complete

Added Footer with link to TVMaze and re-sized images in cards.

kmh256 committed:
Added Search Bar

Added Search Bar & formatting.

kmh256 committed:
Added Episode Selector

Added Episode Selector

kmh256 committed:
Updated formatting for Episodes

Changed formatting to account for seasons and episodes >9.

kmh256 committed:
Updated Episode selector to show all episodes

Episode selector updated to show all episodes

kmh256 committed:
Changed variable name for all episode selector

Changed variable name for all episode selector to selectAllEpisodes

kmh256 committed:
Updated layout and episode counter

Updated layout of header and made episode counter change when single episode selected.

kmh256 committed:
Added search bar

Added search bar

kmh256 committed:
Removed search button and updated layout

Removed search button and updated layout of header

kmh256 committed:
Fixed search bar

Search bar now updates page so only cards that include the search string are displayed.

kmh256 committed:
Level 350 Complete

Used fetch() to get JSON GoT content from the TVMaze API

kmh256 committed:
Added Show Selector

Added Show Selector and adjusted spacing for form.

kmh256 committed:
Level 400 in progress

Level 400 in progress

kmh256 committed:
Fixed some issues with search function

Fixed some issues with search function and variables

AhmadJanah reviewed (commented):
Hi Kara,

I reviewed your code and it looks amazing. I liked your style and colors. I liked your using of css. You did a great job.
However, You seem you are stuck in some points as you didn't hide the previous divs and and didn't go further in the levels. Anyway, if you need any help or support, please don't hesitate to ask.
to summarise, you did great until now. Your code is organised, and I am ready if you have any queries.

Regards,

DanGhanbari commented on script.js:
```diff
@@ -1,12 +1,261 @@
+        counter.textContent = `Displaying ${allShows.length}/240 show(s)`;
+      }
+    }
+    // showDisplay();
```
I have noticed you created a function and never called it. I just noticed you commented out that function.

DanGhanbari commented on script.js:
```diff
@@ -1,12 +1,261 @@
+  searchBar.className = "form-control col-3 ml-4 mt-2 mr-2 mb-3";
+  searchBar.addEventListener("keyup", (event) => {
+    const searchString = event.target.value.toLowerCase();
+    const episodeCards = document.querySelectorAll(".episode-card");
```
You already defined this variable in global scope in line 27.

artcodespace replied:
As Dan said you've already declared this variable. It doesn't actually affect how this is working, because you never refer to the global variable. I'd recommend getting rid of the global declaration. Everything will still work because all of the other references to it are block scoped

DanGhanbari reviewed (commented):
Good job Kara on your TV-Project. I have noticed a couple of minor mistakes and left some comments.

kmh256 committed:
Fixed show and episode selectors

Shows are now hidden when a series is selected. Previous episodes are removed when another series is selected.

kmh256 committed:
Fixed issue with counter and episode drip down

Counter now displays correct number of shows / episodes displayed. Drop down for episodes appears when show selected.

artcodespace commented on index.html:
```diff
@@ -3,19 +3,30 @@
     <!-- Loads a provided function called getAllEpisodes() which returns all episodes -->
-    <script src="episodes.js"></script>
-
+    <!-- <script src="episodes.js"></script> -->
```
Appreciate you may still be working on this but try to get into the habit of removing commented code

artcodespace commented on script.js:
```diff
@@ -1,12 +1,293 @@
+        // Removes episode cards
+        for (let i = 0; i < allEpisodeCards.length; i++) {
+          allEpisodeCards[i].remove();
+        }
```
This for loop is repeated three times between lines 62 and here. Could you think of a way of running this loop once, and putting the `if` logic inside it?

artcodespace commented on script.js:
```diff
@@ -1,12 +1,293 @@
+  function getShowDataFromApi(api) {
+    fetch(api)
+    .then((response) => {
+      if (response.status >= 200 && response.status <= 299) {
```
Apparently there's a property `response.ok` that tells you if the response is ok or not, to save you having to add manual checks like this

artcodespace commented on script.js:
```diff
@@ -1,12 +1,293 @@
+        countShows ++;
+      }
+    }
+    counter.textContent = `Displaying ${countShows}/240 show(s)`;
```
Could you think of how to remove the hardcoding of the number `240` from here?

artcodespace commented on script.js:
```diff
@@ -1,12 +1,293 @@
+    if (episode.season <= 9) {
+      episodeSeason = `0${episode.season}`;
+    }
+    if (episode.number <= 9) {
+      episodeNumber = `0${episode.number}`;
+    }
```
Duplicate of the three lines above

artcodespace commented on script.js:
```diff
@@ -1,12 +1,293 @@
+
+    let episodeOptionTag = document.createElement("option");
+    episodeOptionTag.value = episode.id;
+    let episodeSeason = episode.season;
+    let episodeNumber = episode.number;
```
An option to avoid extra declarations like this would be to destructure on line 218. Then you could just use `season`, `number` etc

artcodespace reviewed (commented):
Left some comments on the code for you to have a look at Kara. Feel free to message me if any of the notes don't make sense to you.

kmh256 committed:
Reduced number of loops used

Reduced repeated for loops

kmh256 committed:
Sorted shows alphabetically in drop-down menu

Sorted shows alphabetically in drop-down menu
